### PR TITLE
Improve the search for clang;

### PR DESCRIPTION
**Pull request by joergdw**

### CODE REVIEW

**Comment by jpalermo** [comment]
Is the old way causing problems somewhere @joergdw? Not that the change isn't better, just wondering what situation triggered the change.

File: packages/ruby-2.7/packaging [hunk 1]
```diff
@@ -15,7 +15,7 @@ if [ $(uname -s) == "Darwin" ]; then
 fi
 
 # Use Clang if available; the resulting Ruby is faster
-if [ -x /usr/bin/clang ]; then
+if command -v clang &> /dev/null; then
   export CXX=/usr/bin/clang++ CC=/usr/bin/clang
```
Comment (bgandon) [inline packages/ruby-2.7/packaging]: Whenever we find `clang` anywhere else than `/usr/bin`, why would we infer there is one in `/usr/bin`?

Seems the env var setup is not aligned with the actual location where clang is found.

File: packages/ruby-2.7/packaging [hunk 2]
```diff
@@ -15,7 +15,7 @@ if [ $(uname -s) == "Darwin" ]; then
 fi
 
 # Use Clang if available; the resulting Ruby is faster
-if [ -x /usr/bin/clang ]; then
+if command -v clang &> /dev/null; then
```
Comment (bgandon) [inline packages/ruby-2.7/packaging]: Where could `clang` be located, if not in `/usr/bin`?

File: packages/ruby-2.7/packaging [hunk 3]
```diff
@@ -32,7 +32,7 @@ tar xzf "yaml-${LIBYAML_VERSION}.tar.gz"
   set -e
   cd "yaml-${LIBYAML_VERSION}"
 
-  
+
```
Comment (bgandon) [inline packages/ruby-2.7/packaging]: Please cleanup

**Review by bgandon — CHANGES_REQUESTED** [review]

File: packages/ruby-2.7/packaging (hunk 1 above)
Comment (joergdw) [reply packages/ruby-2.7/packaging]: Thank you! Fix done!

**Comment by joergdw** [comment]
Yes, it is causing problems, see <https://github.com/cloudfoundry/bosh-package-ruby-release/issues/36>. In our case it is located elsewhere, defined by `PATH`.

File: packages/ruby-2.7/packaging [hunk 4]
```diff
@@ -15,8 +15,14 @@ if [ $(uname -s) == "Darwin" ]; then
 fi
 
 # Use Clang if available; the resulting Ruby is faster
-if [ -x /usr/bin/clang ]; then
-  export CXX=/usr/bin/clang++ CC=/usr/bin/clang
+if command -v clang &> /dev/null
```
Comment (joergdw) [inline packages/ruby-2.7/packaging:18]: Using `command -v` works on any bash-compliant shell. It is a builtin. The most precise solution would be one that is robust to aliases, such as `which --skip-alias clang` but this assumes `which` to be installed in a sufficiently new version.

File: packages/ruby-2.7/packaging [hunk 5]
```diff
@@ -15,8 +15,14 @@ if [ $(uname -s) == "Darwin" ]; then
...
+then
+	export CC="$(command -v clang)"
```
Comment (joergdw) [inline packages/ruby-2.7/packaging:20]: Using `command -v` works on any bash-compliant shell. It is a builtin. The most precise solution would be one that is robust to aliases, such as `which --skip-alias clang` but this assumes `which` to be installed in a sufficiently new version.

**Comment by jpalermo** [comment]
Hi @joergdw this looks good now, however could you make the changes to `ci/templates/packages/ruby/packaging.erb` and `ci/templates/src/compile.env.erb` too? Those are actually the templates that are used to generate the scripts.

**Comment by joergdw** [comment]
Thank you for that comment @jpalermo. I have adapted the two templates you have mentioned.

**Review by aramprice — APPROVED** [review]

**Review by xtreme-nitin-ravindran — APPROVED** [review]

**Comment by rkoster** [comment]
Thanks! @joergdw

**Pull request merged**